client/my-sites/customer-home/mastering-gutenberg-card/index.jsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import { localize } from 'i18n-calypso';
+import { Button, Dialog } from '@automattic/components';
+import { isMobile } from '@automattic/viewport';
+
+/**
+ * Internal dependencies
+ */
+import DismissibleCard from 'blocks/dismissible-card';
+import CardHeading from 'components/card-heading';
+import Gridicon from 'components/gridicon';
+import { localizeUrl } from 'lib/i18n-utils';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class MasteringGutenbergCard extends Component {
+	state = {
+		showDialog: false,
+	};
+
+	handleDialogClosure = () => {
+		this.setState( {
+			showDialog: false,
+			adjustingBlocks: false,
+			navigatingBlocks: false,
+			addingLinks: false,
+			usingImages: false,
+		} );
+	};
+
+	handleAdjustingBlocksClick = () => {
```

Comment:
I think we can further take this to just one handler instead of four, since the only difference between them is a string identifier for the button clicked (which we can [pass as an argument](https://reactjs.org/docs/handling-events.html)). So something like:

```js
handleButtonClick = ( button ) => {
    this.setState( { showDialog: true, activeDialog: button } );
}
```

Then each button would have:

```js
onClick={ this.handleButtonClick.bind( this, 'addingLinks' ) }
```

Reply:
Yes, that works perfectly and is much cleaner, thank you!